neural/predict_with_neural.py

```diff
@@ -0,0 +1,227 @@
+    # from the second notebook
+
+    bertembeder = BertEmbedder(bertmodel, bert_config)
+    fdtype = torch.float32
+
+    # generate and dump bert embedding for the tokens inside the specificed embedding directory
+    bert_proc_docs = generate_sents_embeds_from_docs(docs_data_tensor, bertembeder, sents_embed_path, fdtype,
+                                                     gpu_index=gpu_index)
+    ReaderWriter.dump_data(bert_proc_docs, os.path.join(sents_embed_path, 'bert_proc_docs.pkl'))
+
+    return sents_embed_path
+
+
+def biobert_predict(data_dict: dict, questions):
+    """
+    Make an autoDiscern prediction for an article data_dict using the HEA BioBERT model. Includes all of the data
+    preprocessing steps as were applied for the training of the HEA BioBERT model.
+    Args:
+        data_dict: dictionary of {id: sib-dict}, with sub-dictionary with keys ['url', 'content', 'id', 'responses']
+    Returns: autodiscern predictions for the article.
+    """
+
+    base_dir = BASE_DIR
+    to_gpu = TO_GPU
+    gpu_index = GPU_INDEX
+    question_fold_map = QUESTION_FOLD_MAP
+    experiment_dir = BIOBERT_EXP_DIR
+    working_dir = 'predict'
+
+    vocab_path = os.path.join(base_dir, 'aa_neural/aws_downloads/bert-base-cased-vocab.txt')
+    processor_config = {'tokenizer_max_sent_len': 300,
+                        'label_cutoff': 3,
+                        'label_avgmethod': 'round_mean'}
+
+    sents_embed_dir = os.path.join(base_dir, 'aa_neural/sents_bert_embed_cased')
+    bert_config = {'bert_train_flag': False,
+                   'bert_all_output': False}
+
+    state_dict_path_form = 'train_validation/question_{}/fold_{}/model_statedict/'
+    config_path_form = 'test/question_{}/fold_0/config/'
+
+    default_device = get_device(to_gpu=False)
+
+    # ---
+
+    q_partitions = create_prediction_qdoc_partitions(questions, question_fold_map)
+
+    # run data processing
+    # USED "2019-05-02_15-49-09_a0745f9_sent_level_MM.pkl"
+    html_to_sentence_transformer = adt.Transformer(leave_some_html=True,
+                                                   html_to_plain_text=True,
+                                                   segment_into='sentences',
+                                                   flatten=True,
+                                                   # remove_newlines=False,  # in newer version
+                                                   annotate_html=True,
+                                                   parallelism=False)
+    transformed_data = html_to_sentence_transformer.apply(data_dict)
+
+    # load BERT model
+    pytorch_dump_path = os.path.join(base_dir, 'aa_neural', 'pytorch_biobert')
+    bert_for_pretrain = load_biobert_model(pytorch_dump_path, default_device)
+    bertmodel = bert_for_pretrain.bert
+
+    processor = build_DataDictProcessor(transformed_data, vocab_path, processor_config)
+    tokenizer = BertTokenizer.from_pretrained(vocab_path, do_lower_case=True)  # on LeoMed
```

Comment:
`do_lower_case` should be `False` since you are using `BioBERT`. As a reminder, in our setup `BERT` uses `uncased` and `BioBERT` uses `cased`

Reply:
We could have used `cased` for `BERT` too but we opted originally to use `uncased` (nothing special here)